websockify/token_plugins.py

```diff
@@ -173,18 +190,36 @@ class TokenRedis():
     """
     def __init__(self, src):
         try:
-            # import those ahead of time so we provide error earlier
             import redis
-            self._server, self._port = src.split(":")
-            self._client = redis.Redis(host=self._server, port=self._port)
+        except ImportError:
+            logger.error("Unable to load redis module")
+            sys.exit()
+        # Default values
+        self._port = 6379
+        self._db = 0
+        self._password = None
+        try:
+            fields = src.split(":")
+            if len(fields) == 1:
+                self._server = fields[0]
+            elif len(fields) == 2:
+                self._server, self._port = fields
+            elif len(fields) == 3:
+                self._server, self._port, self._db = fields
+            elif len(fields) == 4:
+                self._server, self._port, self._db, self._password = fields
```

Comment:
Have you considered support if people want the default value for something "in the middle"? E.g. `host.example.com::dbname` would still use the default port.

Reply:
Added support for this in the latest commit.